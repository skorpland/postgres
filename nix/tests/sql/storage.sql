-- storage schema owner
select
  n.nspname as schema_name,
  r.rolname as owner
from
  pg_namespace n
join
  pg_roles r on n.nspowner = r.oid
where
  n.nspname = 'storage';

-- storage schema tables with owners and rls policies
select
  ns.nspname as schema_name,
  c.relname as table_name,
  r.rolname as owner,
  c.relrowsecurity as rls_enabled,
  string_agg(p.polname, ', ' order by p.polname) as rls_policies
from
  pg_class c
join
  pg_namespace ns on c.relnamespace = ns.oid
join
  pg_roles r on c.relowner = r.oid
left join
  pg_policy p on p.polrelid = c.oid
where
  ns.nspname = 'storage'
  and c.relkind = 'r'
group by
  ns.nspname, c.relname, r.rolname, c.relrowsecurity
order by
  c.relname;

-- storage indexes with owners
select
  ns.nspname as table_schema,
  t.relname as table_name,
  i.relname as index_name,
  r.rolname as index_owner
from
  pg_class t
join
  pg_namespace ns on t.relnamespace = ns.oid
join
  pg_index idx on t.oid = idx.indrelid
join
  pg_class i on idx.indexrelid = i.oid
join
  pg_roles r on i.relowner = r.oid
where
  ns.nspname = 'storage'
order by
  t.relname, i.relname;

-- storage schema functions with owners
select
  n.nspname as schema_name,
  p.proname as function_name,
  r.rolname as owner
from
  pg_proc p
join
  pg_namespace n on p.pronamespace = n.oid
join
  pg_roles r on p.proowner = r.oid
where
  n.nspname = 'storage'
order by
  p.proname;

-- storage service migrations
select * from storage.migrations ;
