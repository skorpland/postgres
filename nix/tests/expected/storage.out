-- storage schema owner
select
  n.nspname as schema_name,
  r.rolname as owner
from
  pg_namespace n
join
  pg_roles r on n.nspowner = r.oid
where
  n.nspname = 'storage';
 schema_name |     owner      
-------------+----------------
 storage     | powerbase_admin
(1 row)

-- storage schema tables with owners and rls policies
select
  ns.nspname as schema_name,
  c.relname as table_name,
  r.rolname as owner,
  c.relrowsecurity as rls_enabled,
  string_agg(p.polname, ', ' order by p.polname) as rls_policies
from
  pg_class c
join
  pg_namespace ns on c.relnamespace = ns.oid
join
  pg_roles r on c.relowner = r.oid
left join
  pg_policy p on p.polrelid = c.oid
where
  ns.nspname = 'storage'
  and c.relkind = 'r'
group by
  ns.nspname, c.relname, r.rolname, c.relrowsecurity
order by
  c.relname;
 schema_name | table_name |         owner          | rls_enabled | rls_policies 
-------------+------------+------------------------+-------------+--------------
 storage     | buckets    | powerbase_storage_admin | f           | 
 storage     | migrations | powerbase_storage_admin | f           | 
 storage     | objects    | powerbase_storage_admin | t           | 
(3 rows)

-- storage indexes with owners
select
  ns.nspname as table_schema,
  t.relname as table_name,
  i.relname as index_name,
  r.rolname as index_owner
from
  pg_class t
join
  pg_namespace ns on t.relnamespace = ns.oid
join
  pg_index idx on t.oid = idx.indrelid
join
  pg_class i on idx.indexrelid = i.oid
join
  pg_roles r on i.relowner = r.oid
where
  ns.nspname = 'storage'
order by
  t.relname, i.relname;
 table_schema | table_name |     index_name      |      index_owner       
--------------+------------+---------------------+------------------------
 storage      | buckets    | bname               | powerbase_storage_admin
 storage      | buckets    | buckets_pkey        | powerbase_storage_admin
 storage      | migrations | migrations_name_key | powerbase_storage_admin
 storage      | migrations | migrations_pkey     | powerbase_storage_admin
 storage      | objects    | bucketid_objname    | powerbase_storage_admin
 storage      | objects    | name_prefix_search  | powerbase_storage_admin
 storage      | objects    | objects_pkey        | powerbase_storage_admin
(7 rows)

-- storage schema functions with owners
select
  n.nspname as schema_name,
  p.proname as function_name,
  r.rolname as owner
from
  pg_proc p
join
  pg_namespace n on p.pronamespace = n.oid
join
  pg_roles r on p.proowner = r.oid
where
  n.nspname = 'storage'
order by
  p.proname;
 schema_name | function_name |         owner          
-------------+---------------+------------------------
 storage     | extension     | powerbase_storage_admin
 storage     | filename      | powerbase_storage_admin
 storage     | foldername    | powerbase_storage_admin
 storage     | search        | powerbase_storage_admin
(4 rows)

-- storage service migrations
select * from storage.migrations ;
 id | name | hash | executed_at 
----+------+------+-------------
(0 rows)

